Emby.Server.Implementations/Data/SqliteItemRepository.cs

```diff
@@ -2946,7 +2947,7 @@ private string GetOrderByText(InternalItemsQuery query)
                 if (hasSimilar)
                 {
                     prepend.Add(("SimilarityScore", SortOrder.Descending));
-                    prepend.Add((ItemSortBy.Random, SortOrder.Ascending));
+                    prepend.Add((ItemSortBy.SortName, SortOrder.Ascending));
```

Comment:
Prevent `SimilarityScore` from needing to join with user data. Alternatively, I could add one more check to `EnableJoinUserData`.